tests/foreman/performance/test_candlepin_concurrent_subscription_ak.py

```diff
@@ -0,0 +1,172 @@
+"""Test class for concurrent subscription by Activation Key"""
+from datetime import date
+from robottelo.common import conf
+from robottelo.cli.activationkey import ActivationKey
+from robottelo.cli.factory import (make_activation_key)
+from robottelo.cli.subscription import Subscription
+
+from robottelo.performance.perf_concurrent_test import ConcurrentTestCase
+
+
+class ConcurrentSubActivationKeyTestCase(ConcurrentTestCase):
+    """Concurrently Subscribe to Satellite Server by activation-key tests"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(ConcurrentSubActivationKeyTestCase, cls).setUpClass()
+
+        # parameters for concurrent activation key test
+        cls.savepoint = conf.properties[
+            'performance.test.savepoint2_enabled_repos']
+        cls.raw_file_name = conf.properties[
+            'performance.csv.raw_ak_file_name']
+        cls.stat_file_name = conf.properties[
+            'performance.csv.stat_ak_file_name']
+
+        # add date to csv files names
+        today = date.today()
+        cls.raw_file_name = '{0}-{1}'.format(
+            today.strftime('%Y%m%d'), cls.raw_file_name)
+        cls.stat_file_name = '{0}-{1}'.format(
+            today.strftime('%Y%m%d'), cls.stat_file_name)
+
+    def setUp(self):
+        super(ConcurrentSubActivationKeyTestCase, self).setUp()
+
+        # Restore database before concurrent subscription
+        self._restore_from_savepoint(self.savepoint)
+
+        # Create activation key
+        self.logger.info('---------------------------------')
+        self.logger.info('Create activation key: ')
+        self.logger.info('---------------------------------')
+        (ak_id, ak_name) = self._create_activation_key()
+
+        # Get subscription id
+        self.logger.info('---------------------------------')
+        self.logger.info('Get subscription id: ')
+        self.logger.info('---------------------------------')
+        (sub_id, sub_name) = self._get_subscription_id()
+        self.sub_id = sub_id
+
+        # Add activation key to subscription
+        self._add_ak_to_subscription(ak_id, sub_id)
+
+    def _create_activation_key(self):
+        """Create a new activation key named ak-1"""
+        make_activation_key({
+            'organization-id': self.org_id,
+            'content-view': self.content_view,
+            'lifecycle-environment': self.life_cycle_env,
+            'name': self.ak_name
+        })
+
+        # output activation key informatin
+        self.logger.info('Retrieve activation keys info list:')
+        result = ActivationKey.list({
+            'organization-id': self.org_id},
+            per_page=False
+        )
```

Comment:
Place each argument on its own line whenever reasonable.

```python
        result = ActivationKey.list(
            {'organization-id': self.org_id},
            per_page=False,
        )
```

Using `({` is a good idea when you have *one* argument, it is a dict, and it
won't fit on one line.

```python
some_method({
    'one': 'single',
    'dict': 'argument (too long for one line)',
    'foo': 'bar',
    'biz': 'baz',
})
```